modules/core/keeper/msg_server.go

```diff
@@ -807,7 +810,7 @@ func (k Keeper) ChannelUpgradeTry(goCtx context.Context, msg *channeltypes.MsgCh
 		ctx.Logger().Error("channel upgrade try callback failed", "port-id", msg.PortId, "channel-id", msg.ChannelId, "error", err.Error())
 
 		// we are not returning the error here because we want to commit the error receipt to state
-		if err := k.ChannelKeeper.RestoreChannelAndWriteErrorReceipt(ctx, msg.PortId, msg.ChannelId, upgradeSequence, err); err != nil {
+		if err := k.ChannelKeeper.RestoreChannel(ctx, msg.PortId, msg.ChannelId); err != nil {
```

Comment:
[q] do we want to write an error receipt in the case of an application call back error? This new code as it currently is will just do a restore but not write an error receipt to state.

Reply:
I believe [according to the spec](https://github.com/cosmos/ibc/blob/main/spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md#utility-functions) also an error receipt would be written in this scenario. cc: @AdityaSripal